### PR TITLE
Validate Pack Dependencies

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: demisto/etc/issues/26064

## Description
* Added the *--no-update* flag to the **find-dependencies** command.
*  Added pack dependencies validation.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 1]
```diff
@@ -907,14 +946,21 @@ def build_dependency_graph(pack_id, id_set, verbose_file):
         return graph
 
     @staticmethod
-    def find_dependencies(pack_name, id_set_path='', debug_file_path=''):
+    def find_dependencies(pack_name, id_set_path='', debug_file_path='', echo_results=True,
```
Comment (yaakovi) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:949]: should be ordered by importance:
* pack_name
* id_set_path
* exclude_ignored_dependencies
* update_pack_metadata
* echo_results - switch to `silent_mode` / `quiet_mode`
* debug_file_path

File: demisto_sdk/commands/validate/validate_manager.py [hunk 2]
```diff
@@ -386,6 +389,7 @@ def run_validation_using_git(self):
         validation_results.add(self.validate_added_files(added_files, modified_files))
         validation_results.add(self.validate_changed_packs_unique_files(modified_files, added_files,
                                                                         changed_meta_files))
+        validation_results.add(self.validate_pack_dependencies(modified_files, added_files))
```
[the remark this answers is not in the document]
Comment (Bargenish) [reply demisto_sdk/commands/validate/validate_manager.py:392]: there is no one to pass. the validate command receives an id_set to validate and not an id_set for running the commands.
would you like me to add it as an option for the command?

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 3]
```diff
@@ -198,19 +201,22 @@ def _search_packs_by_items_names_or_ids(items_names, items_list):
                 item_details = list(item_from_id_set.values())[0]
                 if (machine_name in item_possible_names or item_name == item_details.get('name')) \
                         and item_details.get('pack'):
-                    if item_details.get('pack') not in constants.IGNORED_DEPENDENCY_CALCULATION:
+                    if exclude_ignored_dependencies and item_details.get('pack') not in \
+                            constants.IGNORED_DEPENDENCY_CALCULATION:
+                        packs.add(item_details.get('pack'))
+                    else:
                         packs.add(item_details.get('pack'))
```
[the remark this answers is not in the document]
Comment (Bargenish) [reply demisto_sdk/commands/find_dependencies/find_dependencies.py:208]: only if the `exclude_ignored_dependencies` set to `False`

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -921,6 +922,11 @@ def invalid_package_structure(invalid_files):
                'https://github.com/demisto/content/tree/master/docs/package_directory_structure. ' \
                'The files are:\n{}'.format('\n'.join(list(invalid_files)))
 
+    @staticmethod
+    @error_code_decorator
+    def invalid_package_dependencies(pack_name):
+        return f'{pack_name} depends on NonSupported / Deprecated packs'
```
Comment (yaakovi) [inline demisto_sdk/commands/common/errors.py:928]:
```suggestion
        return f'{pack_name} depends on NonSupported / DeprecatedContent packs.'
```

File: demisto_sdk/commands/validate/validate_manager.py [hunk 5]
```diff
@@ -581,6 +586,46 @@ def validate_modified_files(self, modified_files):
...
+
+        dependency_result = json.dumps(first_level_dependencies, indent=4)
+        click.echo(click.style(f"Found dependencies result for {pack} pack:", bold=True))
+        click.echo(click.style(dependency_result, bold=True))
+
+        if first_level_dependencies.get('NonSupported') or first_level_dependencies.get('DeprecatedContent'):
+            error_message, error_code = Errors.invalid_package_dependencies(pack)
+            if self.handle_error(error_message, error_code, file_path=pack):
+                return False
+            return False
```
Comment (yaakovi) [inline demisto_sdk/commands/validate/validate_manager.py:608]: if `handle_error` return false, it means the error should be ignored.
```suggestion
            if self.handle_error(error_message, error_code, file_path=pack):
                return False
```

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 6]
```diff
@@ -907,27 +946,38 @@ def build_dependency_graph(pack_id, id_set, verbose_file):
         return graph
 
     @staticmethod
-    def find_dependencies(pack_name, id_set_path='', debug_file_path=''):
+    def find_dependencies(pack_name, id_set_path='', exclude_ignored_dependencies=True, update_pack_metadata=True,
+                          silent_mode=True, debug_file_path=''):
```
Comment (yaakovi) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:950]:
```suggestion
                          silent_mode=False, debug_file_path=''):
```

File: demisto_sdk/commands/find_dependencies/find_dependencies.py [hunk 7]
```diff
@@ -907,27 +946,38 @@ def build_dependency_graph(pack_id, id_set, verbose_file):
...
+                exclude_ignored_dependencies=exclude_ignored_dependencies)
         first_level_dependencies, _ = parse_for_pack_metadata(dependency_graph, pack_name)
-        update_pack_metadata_with_dependencies(pack_name, first_level_dependencies)
-        # print the found pack dependency results
-        click.echo(click.style(f"Found dependencies result for {pack_name} pack:", bold=True))
-        dependency_result = json.dumps(first_level_dependencies, indent=4)
-        click.echo(click.style(dependency_result, bold=True))
+        if update_pack_metadata:
+            update_pack_metadata_with_dependencies(pack_name, first_level_dependencies)
+        if silent_mode:
```
Comment (yaakovi) [inline demisto_sdk/commands/find_dependencies/find_dependencies.py:978]:
```suggestion
        if not silent_mode:
```

File: demisto_sdk/commands/common/hook_validations/pack_unique_files.py [hunk 8]
```diff
@@ -275,5 +282,34 @@ def validate_pack_unique_files(self):
...
         return self.get_errors()
+
+    # pack dependencies validation
+    def validate_pack_dependencies(self, id_set_path=None):
+        click.secho(f'\n================= Running pack dependencies validation on {self.pack}=================',
+                    fg="bright_cyan")
+        core_pack_list = tools.get_remote_file('Tests/Marketplace/core_packs_list.json') or []
+
+        first_level_dependencies = PackDependencies.find_dependencies(
+            self.pack, id_set_path=id_set_path, silent_mode=False, exclude_ignored_dependencies=False,
```
Comment (yaakovi) [inline demisto_sdk/commands/common/hook_validations/pack_unique_files.py:297]:
```suggestion
            self.pack, id_set_path=id_set_path, silent_mode=True, exclude_ignored_dependencies=False,
```

**Pull request merged**